types/src/transaction/transaction.rs

```diff
@@ -200,6 +200,77 @@ pub struct AssetWrapCCCOutput {
     pub amount: u64,
 }
 
+impl Order {
+    // FIXME: Remove this after the clippy nonminimal bool bug is fixed
+    // https://rust-lang.github.io/rust-clippy/v0.0.212/#nonminimal_bool
+    #![cfg_attr(feature = "cargo-clippy", allow(clippy::nonminimal_bool))]
+    pub fn verify(&self) -> Result<(), Error> {
+        // If asset_amount_fee is zero, it means there's no fee to pay.
+        if self.asset_type_from == self.asset_type_to
+            || self.asset_amount_fee != 0
+                && (self.asset_type_from == self.asset_type_fee || self.asset_type_to == self.asset_type_fee)
+        {
+            return Err(Error::InvalidOrderAssetTypes {
+                from: self.asset_type_from,
+                to: self.asset_type_to,
+                fee: self.asset_type_fee,
+            })
+        }
+        if (self.asset_amount_from == 0) ^ (self.asset_amount_to == 0) {
```

Comment:
When is asset_amount_from and asset_amount_to both 0?

Reply:
When all the order is spent.